karapace/schema_models.py

```diff
@@ -76,23 +82,40 @@ def __init__(self, schema_type: SchemaType, schema_str: str):
             schema_str (str): The original schema string
         """
         self.schema_type = schema_type
-        self.schema_str = schema_str
-        self.max_id: Optional[int] = None
+        self.schema_str = TypedSchema.normalize_schema_str(schema_str, schema_type)
+        self.max_id: Optional[SchemaId] = None
 
         self._str_cached: Optional[str] = None
```

Comment:
Not used anymore, can be removed.
Every schema string is normalized at the constructor.

Reply:
Removed `_str_cached`.